src/Misc/Part.cpp

```diff
@@ -1038,6 +1050,8 @@ int Part::saveXML(const char *filename)
     xml.endbranch();
 
     int result = xml.saveXMLfile(filename, gzip_compression);
+    strncpy(loaded_file,filename, sizeof(loaded_file));
```

Comment:
Like I mentioned in the now outdated comment, if the length of `filename` will be equally long to/longer than `sizeof(loaded_file)`, this will result in a non-null-terminated string and can lead to crashes. See the `strncpy` man page.